EMA/src/app/pages/provider-profile/provider-profile.page.ts

```diff
@@ -0,0 +1,47 @@
+import {Component, OnInit} from '@angular/core';
+
+@Component({
+    selector: 'app-provider-profile',
+    templateUrl: './provider-profile.page.html',
+    styleUrls: ['./provider-profile.page.scss'],
+})
+
+export class ProviderProfilePage implements OnInit {
+
+    tempOwnerDescription;
+    ownerDescription;
+    tempServiceDescription;
+    serviceDescription;
+    clickedOwner = false;
+    clickedService = false;
+    ownerButtonContent: string;
+    serviceButtonContent: string;
+
+    constructor() {
+    }
+
+    ngOnInit() {
+        this.ownerButtonContent = 'Edit';
+        this.serviceButtonContent = 'Edit';
+    }
+
+    editOwner() {
+        this.clickedOwner = !this.clickedOwner;
+        if (this.ownerButtonContent === 'Edit') {
+            this.ownerButtonContent = 'Save';
+        } else if (this.ownerButtonContent === 'Save') {
+            this.ownerButtonContent = 'Edit';
+            this.ownerDescription = this.tempOwnerDescription;
+        }
+    }
+
```

Comment:
do we use the profile-handler here? (for adding details from the profile to the db)